internal/future/statestore/redis.go

```diff
@@ -74,12 +78,17 @@ func newRedis(cfg config.View) (Service, error) {
 	// Encountered an issue getting a connection from the pool.
 	if err != nil {
 		redisLogger.WithFields(logrus.Fields{
-			"error": err.Error(),
-			"query": "SELECT 0"}).Error("state storage connection error")
+			"cmd":   "SELECT 0",
+			"error": err.Error()}).Error("state storage connection error")
 		return nil, fmt.Errorf("cannot connect to Redis at %s, %s", maskedURL, err)
 	}
 
 	redisLogger.Info("Connected to Redis")
+
+	if cfg.IsSet("redis.expiration") {
+		redisTTL = cfg.GetInt("redis.expiration")
```

Comment:
It's easier to just save the cfg instance on the service and read the TTL per request. That way if there's a config push the new TTL takes.